CardScan/Classes/CreditCardOcr/CreditCardOcrResult.swift

```diff
@@ -35,7 +37,7 @@ public class CreditCardOcrResult: MachineLearningResult {
     }
     
     public static func finishedWithNonNumberSideCard(prediction: CreditCardOcrPrediction, duration: Double, frames: Int) -> CreditCardOcrResult {
-        let result = CreditCardOcrResult(mostRecentPrediction: prediction, number: "", expiry: nil, name: nil, isFinished: true, duration: duration, frames: frames)
+        let result = CreditCardOcrResult(mostRecentPrediction: prediction, number: "", expiry: nil, name: nil, state: .finished, duration: duration, frames: frames)
```

Comment:
While it doesn't directly affect this PR, I would advise against having a single `result` class that can have conflicting values. For example, what would it mean to downstream code if it gets a `CreditCardOcrResult` with a valid number and expiry, but is in state `initial`?

It might make sense to have the `state` contain all the possible valid values for that state. For example, `ocrOnly` would have a method to get the current most likely pan, expiry, etc. while `finished` would have a method to get the final pan, expiry, etc. Initial would have no pan nor expiry.

Reply:
yeah this function is a huge hack -- let's clean it up later